lib/bundler/definition.rb

```diff
@@ -307,6 +307,11 @@ def to_lock
         out << "  #{p}\n"
       end
 
+      if ruby_version
```

Comment:
@segiddins could you please clarify. If `BUNDLED WITH` is already backwards incompatible, what is stopping from adding of this feature then.